openshift/openshift.go

```diff
@@ -0,0 +1,175 @@
+package openshift
```

Comment:
@matejvasek it will be great not to have this inside `func`.. is there any way to keep this out and add things like this as extensions? When I proposed profiles, I was thinking about: 
- registry details
- credentials / CA
These can be created in a generic and non Openshift exclusive way. 

I am not sure about the roundtripper.. maybe we can create a generic roundtripper (first I need to understand what it is doing exactly)

Reply:
The `http.RoundTripper` here is our specialized implementation that can dial access services in cluster.
See: https://github.com/knative-sandbox/kn-plugin-func/blob/main/k8s/dialer_test.go#L26

Reply:
> is there any way to keep this out

This code is called only if `-tags openshift` is used. So upstream build wouldn't use this.

Reply:
>For OpenShift we just need to add the CA to the RoundTripper.

Important: this code trust the OpenShift CA only for limited set of hosts (right now exactly one host -- the registry).

Reply:
wrt profiles:
You mean there would be some custom configuration on filesystem (e.g. in `~/.config/func/`)?

That might be possible, but it implies additional setup that would have to be done user.
(or maybe done by package manager if we had `kn` rpm)

I really would like to make this work without any additional setup.

Reply:
>  is there any way to keep this out and add things like this as extensions?

I believe this is already the case.  The core `func` repo is structured as dependency-free, with third-party dependencies supported through packages which implement their interfaces.  So this is indeed essentially "kept out" and added as an "extension".

This is a common method of structuring repositories to accomplish the goal you are suggesting. 👍🏻